### DISCUSSION TITLE
Emulate LIM EMS 3.0 compatible expanded memory

### DISCUSSION THREAD

**Discussion by Fenix770** [86Box/86Box#4054] [Ideas]
Hello!

I opened this issue as a question of curiosity since I know that the latter Windows 2.0 supports any EMS card with a LIM 4.0 driver.
I found that Windows 1.00 throughout  1.04 appears to support EMS memory. But does not detect or uses the expanded memory of any of the emulated boards. In the Windows install disk there are for EMS drivers files EMM.PC and EMM.AT, they must be for Intel Above Board PC and AT cards. 

I suspect Windows 1.0x only support expanded memory provided by the Intel boards or any board that can emulate the LIM EMS 3.0. (I read in some old pc magazines that some EMS boards were capable to emulate the 3.0 spec.)

The problem seems to be that the LIM EMS 3.2 and 4.0 are not backward compatible with the 3.0 spec.
The only difference with the 3.2 spec I found so far, is, the fuctions 10 and 11 were removed and the maximum RAM memory supported were increased from 4mb to 8mb.

here is a site with some info: [http://nersp.nerdc.ufl.edu/~esi4161/files/dosman/chap10](url)

Question: Is possible to emulate the LIM EMS 3.0 spec with a driver, or one of these cards must be emulated to obtain compatibility with Win 1.0x?

I know both alternatives may be difficult to achieve because of lack of documentation, and also I  had never seen a simulator or emulator that implemet the 3.0 spec.

**Comment by OBattler** [comment]
Is there any documentation for the Intel AboveBoard PC and AT? If yes, I could try to implement them.

**Comment by Fenix770** [comment]
I only found device drivers, but not other relevant information, not even the source code of these drivers in assembler.

**Comment by Torinde** [comment]
[Above Board Plus 8 documentation](https://www.minuszerodegrees.net/manuals/Intel/Intel%20-%20Above%20Board%20Plus%208%20and%20Above%20Board%20Plus%208%20IO%20-%20Installation%20Guide%20-%20May%201989.pdf)
- Conventional memory backfill (to increase it to 640KB for PCs with less)
- Mappable conventional memory (window from inside the first 640KB) - for DESQview, Win2.0
- Expanded memory (window from inside the Upper memory 384KB) - for DOS LIM applications
- up to 14MB per card
- multiple cards, up to 32MB total per PC (what will happen if 4x14MB=56MB are installed?)

[LIM specification](https://web.archive.org/web/20040718235414/http://nersp.nerdc.ufl.edu/~esi4161/files/dosman/chap10)
[Hardware / Intel Above Board Plus](https://retrocmp.de/hardware/above-plus/above-plus.htm)
[Drivers](https://forum.vcfed.org/index.php?threads/intel-above-board-8-bit-drivers.28458/): ftp://ftp.oldskool.org/pub/misc/Hardware/Intel/ ([archive.org](https://web.archive.org/web/20230607130744/ftp://ftp.oldskool.org/pub/misc/Hardware/Intel/))

86Box now has "Intel Above Board" with the following options:
![image](https://github.com/user-attachments/assets/138b44f6-f34b-4d13-9ef9-3cf60b783129)

Related: 
- #4528
- #4922

**Comment by meonhub** [comment]
hi, where on the gui is the intel above board? i can't find it. i'm using 86box v6. thanks

**Reply by lemondrops** [reply]
It's not emulated yet.

**Comment by meonhub** [comment]
ohhh, ok thanks
